client/Components/SingleCohort/SingleCohort.js

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect } from 'react'
+import { connect } from 'react-redux'
+import { fetchCohort, fetchStudents, fetchStretches } from '../../store/thunks'
+import Sidebar from '../Sidebar'
+
+const mapStateToProps = ({ cohort, students, stretches }) => {
+  return {
+    cohort,
+    students,
+    stretches
+  }
+}
+
+const mapDispatchToProps = dispatch => {
+  return {
+    fetchCohort: cohortId => dispatch(fetchCohort(cohortId)),
+    fetchStudents: cohortId => dispatch(fetchStudents(cohortId)),
+    fetchStretches: cohortId => dispatch(fetchStretches(cohortId))
+  }
+}
+
+const SingleCohort = ({
+  match,
+  cohort,
+  students,
+  stretches,
+  fetchCohort,
+  fetchStudents,
+  fetchStretches
+}) => {
+  const cohortId = match.url.slice(8)
```

Comment:
you can just do match.params.id here right? I would avoid using the argument of 8 in slice that is not dynamically generated. That is an example of a magic number or string and they can wreck absolute havoc on code when you make any change

Reply:
Yes absolutely. I'll change this over ASAP. I think match.pararms.id works.

Reply:
fixed - sorry about that. this was bad.

Reply:
No problem at all. I'm only so aware of this b/c of being told what i just told you at work